include/swift/Runtime/Concurrent.h

```diff
@@ -406,6 +409,102 @@ class ConcurrentMap
   }
 };
 
+
+/// An append-only array that can be read without taking locks. Writes
+/// are still locked and serialized, but only with respect to other
+/// writes.
+template <class ElemTy> struct ConcurrentReadableArray {
```

Comment:
This type should be no-assign no-copy no-delete. (I don't remember how those are expressed in Swift's C++ code.)